src/pages/Home/index.tsx

```diff
@@ -68,15 +84,42 @@ const Home: FC = () => {
 	const [userSpecificStreams, setAppStore] = useAppStore((store) => store.userSpecificStreams);
 	const [userRoles] = useAppStore((store) => store.userRoles);
 	const [userAccessMap] = useAppStore((store) => store.userAccessMap);
+	const searchInputRef = useRef<HTMLInputElement>(null);
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredMetaData, setFilteredMetaData] = useState<MetaData>(() => metaData || {});
+
+	useEffect(() => {
+		if (metaData) {
+			setFilteredMetaData(metaData);
+		}
+	}, [metaData]);
 
 	useEffect(() => {
 		if (!Array.isArray(userSpecificStreams) || userSpecificStreams.length === 0) return;
 		getStreamMetadata(userSpecificStreams.map((stream) => stream.name));
 	}, [userSpecificStreams]);
 
+	const filterStreams = useCallback(
+		debounce((term: string) => {
```

Comment:
we should not debounce on data that already there in memory

Reply:
Addressed this